python/rmm/_lib/memory_resource.pyx

```diff
@@ -523,8 +560,14 @@ cdef void* _allocate_callback_wrapper(
     size_t nbytes,
     cuda_stream_view stream,
     void* ctx
-) with gil:
-    return <void*><uintptr_t>((<object>ctx)(nbytes))
+) nogil:
+    cdef CppExcept err
+    with gil:
+        try:
+            return <void*><uintptr_t>((<object>ctx)(nbytes))
+        except BaseException as e:
+            err = translate_python_except_to_cpp(e)
+    throw_cpp_except(err)
```

Comment:
I don't quite understand -- won't the C++ exception thrown here end up _not_ propagating to Python (this function is not marked `except +`)?

Reply:
Yes, thus the caller (a memory resource implemented in C++) can now catch and/or propagate the exception.

Reply:
Ah I see, that makes sense. Thanks!